packages/amplication-server/src/core/workspace/workspace.service.ts

```diff
@@ -694,6 +696,71 @@ export class WorkspaceService {
     return true;
   }
 
+  async dataMigrateWorkspacesResourcesCustomDtos(
+    quantity: number
+  ): Promise<boolean> {
+    const workspaces = await this.prisma.workspace.findMany({
+      where: {
+        projects: {
+          some: {
+            deletedAt: null,
+            resources: {
+              some: {
+                deletedAt: null,
+                archived: { not: true },
+                resourceType: EnumResourceType.Service,
+                blocks: { none: { blockType: EnumBlockType.ModuleDto } },
```

Comment:
do we want "blocks:  { none: ..."  or to remove this condition to allow finding services where not ALL ModuleDTOs were created?
In addition, if someone has ModuleDTOs from custom modules (e.g., Yuval), this will not run the migration for them for the same reason.

I think we should consider how we only find entities that don't have ModuleDTOs, perhaps?